src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1506 @@
+      scale = string.lower(scale)
+      local scale_new = {}
+      -----------------------
+      if scale == "major" or scale == "ionian" then scale_new = {0, 2, 4, 5, 7, 9, 11} end
+      if scale == "dorian" then scale_new = {0, 2, 3, 5, 7, 9, 10}  end
+      if scale == "phrygian" then scale_new = {0, 1, 3, 5, 7, 8, 10} end
+      if scale == "lydian" then scale_new = {0, 2, 4, 6, 7, 9, 11} end
+      if scale == "mixolydian" then scale_new = {0, 2, 4, 5, 7, 9, 10}  end
+      if scale == "natural minor" or scale == "aeolian" then scale_new = {0, 2, 3, 5, 7, 8, 10} end
+      if scale == "harmonic minor" then scale_new = {0, 2, 3, 5, 7, 8, 11} end
+      if scale == "hungarian minor" then scale_new = {0, 2, 3, 6, 7, 8, 11} end
+----- 'Exotic' scales, with something other than 7 scale degrees. Will be treated as chords.
+      if scale == "whole tone" then scale_new = {0, 4, 2, 6, 8, 10} end
+      if scale == "major pentatonic" then scale_new = {0, 4, 7, 2, 9} end
+      if scale == "minor pentatonic" then scale_new = {0, 3, 7, 5, 10} end
+----- Chords - Listed in order of matching priority. 
+      if scale == "dom7" then scale_new = {0, 4, 7, 10, 2, 9, 5} end
+      if scale == "maj7" then scale_new = {11, 0, 4, 7, 9, 2, 5} end
+      if scale == "min7" then scale_new = {0, 3, 7, 10, 2, 5, 9} end
+      if scale == "m7b5" then scale_new = {0, 3, 6, 10, 2, 5, 8} end
+      if scale == "dim7" then scale_new = {0, 3, 6, 9, 2, 5, 8} end
+      if scale == "aug" then scale_new = {0, 4, 8, 2, 10, 6} end
+      --
+
+      --print("New scale (scale_new) numbers are:")
+      for a, b in pairs(scale_new) do
+        scale_new[a] = math.fmod (scale_new[a] + root_off , 12)
+        --print(scale_new[a])
+      end 
+--------------------------
+-- 3) build out scale using letternames
+      local scale_ltrs = root  -- This is where we will build scale (as string of letter-names)
+      local root_string = string.sub(root, 1, 1) -- the base string, A-G
+      local last = root_string -- the last string added to the list
+      local scale_deg = 2 -- will advance through scale. Root was already added, so we will start at 2...
+----
+      if scale == "whole tone" or scale == "major pentatonic" or scale == "minor pentatonic" then
+        chd = true
+      end -- temporary change for exotic scales!
+
+      if not chd then
+        for i = 1, 2, 1 do --- run through this twice...
+          if last == "A"  and scale_deg <= 7 then
+            local found = false
+            for j = 1, 3, 1 do
+              if B_num[j] == scale_new[scale_deg] then
+                scale_ltrs = scale_ltrs..", "..B_ltr[j]
+                found = true
+              end
+            end -- for j
+            if not found then
+              error = true
+              goto error
+            end
+            scale_deg = scale_deg + 1
+            last = "B"
+          end
+          if last == "B"  and scale_deg <= 7 then
+            local found = false
```

Comment:
Guess this review is going to be me being picky about minor code style.

One general rule of thumb with booleans is that the variable should signify that it's a boolean. Often, you'll want to prefix them with `is_` or `should_`, or something else that describes the purpose of the variable. That way, you can read `is_found` and instantly know that variable is a boolean.

It also makes if statements read more like English:

```lua
if is_found then
```